documents/description.md

```diff
@@ -1,9 +1,83 @@
 
 ## Language domain
+The domain of this language is DOTA 2. The user should be able to (in an ideal world) write a program to test any combination of dota 2 heroes, items and skills with as much specificity as possible. It should be impossible to do anything other than test DOTA 2 elements. 
 
 ## Language design
+The language should be designed with three main elements. First, the user should be able to specify a build. This includes:
+ + The hero
+ + The hero's level
+ + The Hero's items
+ + The levels of each ability the hero has
+
+The user can then store builds like variables in a conventional programming language. 
+
+The second element of the language is essentially dota-specific queries. So, one can take one of the builds that is specified, and ask for various stats on that build (for example, damage, hit points, attack speed, DPS, etc). 
+
+Lastly, the language needs control-flow elements. These are essentially ways to compare different builds. Some things that the language might have is a foreach-style loop and list elements, so one can make a list of builds and run the same command on all of them. Additionally, I'd want things that compare two builds. Commands like max(stat, build1,build2), and WhoWins(build1, build2). Obviously, this isn't the syntax, but it describes the general structure of the control flow. 
 
 ## Example computations
+The first kind of computation is the definition of a build, which would look something like:
+```
+build1 = ember spirit level 21 with (boots of travel, battle fury, manta style, crystalis, divine rapier) skills(4,4,4,4,5)
+```
+This would then just store that build as an object in the internal representation. Next, one could do things like (numbers are made up):
+
+```
+build1.damage
+> 600 
+build1.attackSpeed
+> 60
+build1.hitPoints
+> 1800
+```
+
+Now, if one had multiple builds, you could do things like:
+```
+builds = [build1,build2,build3,build4]
+foreach build get damage
+> 600
+> 150
+> 200
+> 300
+```
+Similarly, one could get the max damage build like:
+```
+max(damage, builds)
+> build1: 600
+```
+And finally, comparing two builds would probably work like:
+
+```
+build1 VS build2
+>build1 wins (90% confidence)
+```
+
+These are all three kinds of commands that the language should support.
+
+## Language Evaluation
+Evaluation of the language should follow the pattern of parsing the input, then going to an internal data layer that contains all of the stats for dota heroes and items, and then storing the result as an object in my internal representation. This should internally store all the damage, HP, etc stats in the object, so at this point I never have to go back to the data layer. 
```

Comment:
The "Language Evaluation" section is meant to describe how you'll evaluate the design of the language (not what happens when a program runs)